commit 8b212c18ee2a0fbf1c552d58d2ad36d5651cd3ba
Author: Kelly McLaughlin <kelly.mclaughlin@joyent.com>
Date:   2019-05-15T12:17:07-06:00 (5 months ago)
    
    MANTA-4278 Plumb through object id RPC parameter for createobject function for electric-boray

diff --git a/lib/server.js b/lib/server.js
index 678b11d..d98496d 100644
--- a/lib/server.js
+++ b/lib/server.js
@@ -65,6 +65,7 @@ var CO_ARGS_SCHEMA = [
     { name: 'owner', type: 'string' },
     { name: 'bucket_id', type: 'string' },
     { name: 'name', type: 'string' },
+    { name: 'object_id', type: 'string' },
     { name: 'content_length', type: 'number' },
     { name: 'content_md5', type: 'string' },
     { name: 'content_type', type: 'string' },
@@ -254,7 +255,7 @@ function createServer(options, callback) {
             fast.FAST_CHECKSUM_V1;
 
         log.info('creating boray clients');
-        boray_client.createBucketClient({
+        boray_client.createClient({
             pnodes: opts.dataDirector.getPnodes(),
             borayOptions: options.borayOptions,
             log: options.log,
@@ -782,12 +783,13 @@ function createObject(options) {
         var owner = argv[0];
         var bucket = argv[1];
         var key = argv[2];
-        var content_length = argv[3];
-        var content_md5 = argv[4];
-        var content_type = argv[5];
-        var headers = argv[6];
-        var sharks = argv[7];
-        var props = argv[8];
+        var object_id = argv[3];
+        var content_length = argv[4];
+        var content_md5 = argv[5];
+        var content_type = argv[6];
+        var headers = argv[7];
+        var sharks = argv[8];
+        var props = argv[9];
 
         var id = options.req_id || uuid.v1();
 
@@ -802,7 +804,8 @@ function createObject(options) {
         log.debug({
             owner: owner,
             bucket: bucket,
-            key: key
+            key: key,
+            id: object_id
         }, 'createObject: entered');
 
         options.dataDirector.getObjectLocation(owner, bucket, key,
@@ -822,7 +825,7 @@ function createObject(options) {
                 props = null;
             }
 
-            client.createObject(owner, bucket, key, content_length,
+            client.createObject(owner, bucket, key, object_id, content_length,
                 content_md5, content_type, headers, sharks, props, vnode,
                 function (pErr, meta) {
                 log.debug({
